### DISCUSSION TITLE
failed to recognise redirect URI when using acquire_token_interactive

### DISCUSSION THREAD

**Discussion by shemnor** [AzureAD/microsoft-authentication-library-for-python#494] [Q&A] [answered]
I have added http://localhost/ and http://localhost:62576/ URIs to the azure application but I keep getting an URI mismatch error when using result = app.acquire_token_interactive( scopeInp, port=62576). Am I missing something?

**Comment by shemnor** [comment, marked as answer]
Yes, What I missed was that the clientID was incorrect! Sorry!

**Comment by rayluo** [comment]
Glad that you figured it out of the incorrect client_id.

By the way, [when using localhost as redirect_uri, you do not need to register the port in Azure Portal](https://docs.microsoft.com/en-us/azure/active-directory/develop/reply-url#localhost-exceptions). Out of curiosity, why do you need to specify a specific port on client side in the first place? MSAL Python's `acquire_token_interactive()` will automatically choose an available port for you.

**Reply by shemnor** [reply]
I was registering the port just in case it was the reason why I was having my issue. You are right that the port does not need to be specified. 

Thanks for your help! :)
